include/deal.II/lac/precondition.h

```diff
@@ -905,8 +959,15 @@ class PreconditionChebyshev : public Subscriptor
 
     /**
      * Stores the inverse of the diagonal of the underlying matrix.
+     *
+     * @deprecated Set the variable @p preconditioner defined below instead.
      */
     VectorType matrix_diagonal_inverse;
```

Comment:
There is this `#pragma push/pop` thing (hidden behind a macro) we use to silence warnings in headers. Presumably we could do the same within regular code, but I've never used it. It may be worth a try, though.